library/foreman_provisioner.py

```diff
@@ -0,0 +1,292 @@
+        :rtype: dict
+        """
+        request_url = '{0}/{1}/{2}'.format(self.url, self.default_uri, host_id)
+        response = self.session.get(request_url, verify=False)
+        body = response.json()
+        return body
+
+    def update_host(self, host_id, update_info):
+        """
+        This method updates a host details in foreman
+        :param host_id: the name or ID of the host we wish to update
+        :param update_info: params we wish to update on foreman
+        :type update_info: dict
+        :returns: host information
+        :rtype: dict
+        """
+        request_url = '{0}/{1}/{2}'.format(self.url, self.default_uri, host_id)
+        response = self.session.put(request_url, data=update_info, verify=False)
+        body = response.json()
+        return body.get('host')
+
+    def set_build_on_host(self, host_id, flag):
+        """
+        sets the 'build' flag of a host to a given :param flag:
+        :param host_id: the id or name of the host as it is listed in foreman
+        :param flag: a boolean value (true/false) to set the build flag with
+        """
+        host = self.update_host(host_id, json.dumps({'build': flag}))
+        #TODO(tkammer): add verification that the host was updated
+
+    def bmc(self, host_id, command):
+        """
+        execute a command through the BMC plugin (on/off/restart/shutdown/etc)
+        :param host_id: the id or name of the host as it is listed in foreman
+        :param command: the command to send through the BMC plugin, supported
+        commands: 'status', 'on', 'off', 'cycle', 'reset', 'soft'
+        """
+        request_url = '{0}/{1}/{2}/power'.format(self.url, self.default_uri, host_id)
+        command = json.dumps({'power_action': command})
+        response = self.session.put(request_url, data=command, verify=False)
+        #TODO(tkammer): add verification that the BMC command was issued
+
+    def ipmi(self, host_id, command):
+        """
+        execute a command through the ipmitool
+        :param host_id: the ipmi id of the host
+        :param command: the command to send through the ipmitool
+        commands: 'status', 'on', 'off', 'cycle', 'reset', 'soft' # TBD
+        """
+        command = "ipmitool -I lanplus -H {0} -U ADMIN -P ADMIN chassis power {1}".format(host_id, command)
+        response = os.system(command)
+
+    def _validate_bmc(self, host_id):
+        """
+        This method validate that there is at least one BMC on the given host
+        :param host_id: the id or name of the host as it is listed in foreman
+        """
+        request_url = '{0}/{1}/{2}/interfaces'.format(
+            self.url, self.default_uri, host_id)
+        response = self.session.get(request_url, verify=False)
+        body = response.json()
+        found_bmc = False
+        for interface in body['results']:
+            if "BMC" in interface['type']:
+                found_bmc = True
+                break
+        return found_bmc
```

Comment:
I know that you didn't write it (it was probably me...), but in general, function that returns non 0/False value indicates on some failure.
Please change the return value to be 'not found_bmc' or even change the name of the variable itself to be something with the opposite logic. (and don't forget to change also the 'if' statement which call this function (line 229)